src/blockdev.rs

```diff
@@ -42,6 +42,42 @@ use crate::util::*;
 
 use crate::{runcmd, runcmd_output};
 
+#[derive(Debug, Deserialize)]
+struct DevicesOutput {
+    blockdevices: Vec<Device>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Device {
+    pub name: String,
+    pub label: Option<String>,
+    pub fstype: Option<String>,
+    #[serde(rename = "type")]
+    pub blktype: Option<String>,
+    pub mountpoint: Option<String>,
+    pub uuid: Option<String>,
+    pub children: Option<Vec<Device>>,
+}
+
+impl Device {
+    pub fn lsblk(dev: &Path, with_deps: bool) -> Result<Vec<Device>> {
```

Comment:
So I think here a really nice thing about switching to JSON is that we capture the hierarchy of the block devices. In this case, subdevices of the main device we're querying will be in `children`.

So here I think we want to output a `Result<Device>` instead. The `lsblk` output supports outputting multiple top-level devices, but in our uses of `lsblk` we're always really just interested in a single top-level device (and possibly its children if `with_deps` is enabled). So this function could sanity-check that only one device matched, and return that one.

Reply:
I think we should also rename `with_deps` to `with_children`, since that's the name in the structure we return.

Though hmm, kinda tempting to drop the `with_deps` argument altogether since we'd always return one device now unlike before. `lsblk` will be doing more work than it needs to in some cases, but meh... feels like it's worth the cleaner API we get. @bgilbert might have a different opinion.